app/models/nonprofit.rb

```diff
@@ -180,6 +181,10 @@ def as_json(options = {})
     h
   end
 
+  def state_is_valid
+    errors.add(:state_code, 'must be a US two-letter state code') unless ISO3166::Country[:US].subdivisions.has_key? state_code
```

Comment:
You should probably upcase the state_code like you do in `config/state_code_constraints.rb`

Reply:
You're right, the upcase method should be used. But in order to use it, it's necessary to check if the `state_code` is not nil.

Something like this:

```rb
  def state_is_valid
    if state_code && !ISO3166::Country[:US].subdivisions.has_key?(state_code.upcase)
      errors.add(:state_code, 'must be a US two-letter state code')
    end
  end
```

Reply:
You're on the right track! Ruby actually has a nil-safe navigation operator `&.` which, if you the object on the left is nil, then the whole expression becomes nil. This allows us to keep the search this is a one-liner as follows:

```ruby
errors.add(:state_code, 'must be a US two-letter state code') unless ISO3166::Country[:US].subdivisions.has_key? state_code&.upcase
```